viewing_party/party.py

```diff
@@ -0,0 +1,187 @@
+        if user_data["watchlist"][i]["title"] == title:
+            del user_data["watchlist"][i]
+            break        
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    #define a variable to hold the sum of ratings
+    the_sum = 0
+    #define a variable to get the count of ratings
+    the_count = 0
+    #add the ratings to the sum, and increment the counter
+    try:
+        for record in user_data["watched"]:
+            the_sum += record["rating"]
+            the_count += 1
+
+        #figure out the average by dividing the sum by the count
+        return the_sum/the_count
+    except ZeroDivisionError:
+        return 0
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    if user_data["watched"] != []:
+        #first, create a dict of the genres and how often they appear in the watched list
+        for record in user_data["watched"]:
+            mygenre = record["genre"]
+            genre_dict[mygenre] = genre_dict.get(mygenre,0)+1
+
+
+        #second, find which genre in that newly created dictionary has the greatest value
+        most_popular_genre = ""
+        most_popular_genre_count = 0
+        for k,v in genre_dict.items():
+            if v > most_popular_genre_count:
+                most_popular_genre = k
+        
+        return most_popular_genre
+    else:
+        return None
+
+
+def get_unique_watched(user_data):
+    #Set up a variable to hold the unique titles
+    unique_dict = []
+
+    #Compare each title in watched to the friends watched
+    for record in user_data["watched"]:
+        watched_title = record["title"]
+        watched_title_found = False
+        #If I dont find the title in the friends lists append to new dictionary
+        for friend in user_data["friends"]:
+            for friend_watched_record in friend["watched"]:
+                friend_watched_title = friend_watched_record["title"]
+                if watched_title == friend_watched_title:
+                    watched_title_found = True
+        
+        #I did not find it in the friends, so append it
+        if watched_title_found == False:
+            unique_dict.append(record)
+    
+    return unique_dict
+    
+def get_friends_unique_watched(user_data):
+    #Set up a variable to hold the unique titles
+    unique_dict = []
+    friends_dict = []
+
+    #Loop through all the friends watched data and create a single list of dictionaries
+    #with those records 
+    for friend in user_data["friends"]:
+        for friend_watched_record in friend["watched"]:
+            if friend_watched_record not in friends_dict:
+                friends_dict.append(friend_watched_record)
+
+
+    #Loop through the friends_dict  to compare to user_data_watched
+    for friend_watched_record in friends_dict:
+        if friend_watched_record not in user_data["watched"]:
+            unique_dict.append(friend_watched_record)
+    
+    return unique_dict 
```

Comment:
This isn't the classic helper function case where the code is identical, but this function is doing similar work to `get_unique_watched`. Both functions essentially loop through a list, check if the item is in the other list, and if not, add the item to the result list. Is there a way you could construct a helper function that does this work and construct the lists and pass them to the helper function?